File: sample/src/main/java/com/dylanc/activityresult/launcher/sample/java/launcher/InputTextRequest.java
package com.dylanc.activityresult.launcher.sample.java.launcher;

import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * @author Dylan Cai
 */
public class InputTextRequest {
  @NonNull
  private final String name;
  @NonNull
  private final String title;
  @Nullable
  private final String value;

  public InputTextRequest(@NonNull String name, @NonNull String title, @Nullable String value) {
    this.name = name;
    this.title = title;
    this.value = value;
  }

  @NonNull
  public String getName() {
    return name;
  }

  @NonNull
  public String getTitle() {
    return title;
  }

  @Nullable
  public String getValue() {
    return value;
  }
}

File: sample/src/main/java/com/dylanc/activityresult/launcher/sample/kotlin/launcher/InputTextRequest.kt
/*
 * Copyright (c) 2021. Dylan Cai
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.dylanc.activityresult.launcher.sample.kotlin.launcher

import java.io.Serializable

/**
 * @author Dylan Cai
 */


interface OnFilterValueListener : Serializable {
  fun onFilterValue(value: String): Boolean
}

class InputTextRequest(
  val name: String,
  val title: String = name,
  val hint: String? = null,
  val value: String? = null,
  val listener: OnFilterValueListener? = null
)